bundles/org.openhab.binding.neohub/src/main/java/org/openhab/binding/neohub/internal/NeoHubHandler.java

```diff
@@ -279,4 +391,73 @@ private void fastPollingSchedulerExecute() {
             lazyPollingSchedulerExecute();
         }
     }
+
+    /*
+     * select whether to use the old "deprecated" API or the new API
+     */
+    private void selectApi() {
+        boolean supportsLegacyApi = false;
+        boolean supportsFutureApi = false;
+
+        NeoHubSocket socket = this.socket;
+        if (socket != null) {
+            String responseJson;
+            NeoHubReadDcbResponse systemData;
+
+            try {
+                responseJson = socket.sendMessage(CMD_CODE_READ_DCB);
+                systemData = NeoHubReadDcbResponse.createSystemData(responseJson);
+                supportsLegacyApi = systemData instanceof NeoHubReadDcbResponse;
+            } catch (Exception e) {
```

Comment:
should probably still log this as debug or trace.

Reply:
Ok.